libethcore/Miner.h

```diff
@@ -229,7 +229,7 @@ class Miner: public Worker
 
 	WorkPackage work() const { Guard l(x_work); return m_work; }
 
-	void addHashCount(uint64_t _n) { m_hashCount.fetch_add(_n, std::memory_order_relaxed); }
+	void addHashCount(uint32_t _n) { m_hashCount.fetch_add(_n, std::memory_order_relaxed); }
```

Comment:
Why the change?

Reply:
Called by individual card once per reporting interval. 32 bits will hold up to 4gigqa hashes. Should be adequate for whole rig, and is passed in single 32 bit register.

Reply:
32 bits is enough for 2^32 hashes during reporting interval. Even the fastest card would take nearly 2 minutes to overflow. I'll change it back to 64 bits, as well as the global counter. It is negligible performance penalty.

Reply:
The fastest available card would take a few minutes to overflow, and clminer only passes a 32 bit value is passed in, so there never was a need for 32 bits here.

Reply:
But what was the problem with uint64_t. Switching to uint32_t can make this actually slower.

Reply:
No problem really, but both clminer and cudaminer use 32-bit counters locally, which is then passed as parameter to this function call. I've restored the per miner counter that adds up these successive calls to 64-bits, so no risk of overflow, and the conversion to 64-bit only happens in one place, inside the addHashCount function.

Reply:
The workLoop in both miners use a 32-bit counter to accumulate hash count

Reply:
Nevermind, I just noticed that cudaminer has 64-bit counter locally. I'll change back to 64 bit, and also update clminer to 64 bits